src/component/component.js

```diff
@@ -29,8 +29,18 @@ export let components = {};
 export class Component {
 
     constructor(options = {}) {
-        this.validate(options);
 
+        if (!options.dimensions || !(typeof options.dimensions === 'object')) {
```

Comment:
I think it might be better if we just leave `options.dimensions` blank here. Having a 0 width and height may be confusing for people reading the code in future.